external-import/loaderinsightagency/src/main.py

```diff
@@ -0,0 +1,283 @@
+import stix2
+import yaml
+from dateutil.parser import parse
+from pycti import (
+    Indicator,
+    Malware,
+    OpenCTIConnectorHelper,
+    StixCoreRelationship,
+    get_config_variable,
+)
+
+
+class LIAFileFeed:
+    def __init__(self):
+        config_file_path = os.path.dirname(os.path.abspath(__file__)) + "/config.yml"
+
+        config = (
+            yaml.load(open(config_file_path), Loader=yaml.FullLoader)
+            if os.path.isfile(config_file_path)
+            else {}
+        )
+
+        self.helper = OpenCTIConnectorHelper(config)
+
+        self.api_url = get_config_variable(
+            "LIA_BASE_API_URL", ["liafilefeed", "api_base_url"], config
+        )
+        self.api_key = get_config_variable(
+            "LIA_API_KEY", ["liafilefeed", "api_key"], config
+        )
+
+        self.duration_period = get_config_variable(
+            "CONNECTOR_DURATION_PERIOD", ["connector", "duration_period"], config
+        )
+
+        self.identity = self.helper.api.identity.create(
+            type="Organization",
+            name="Loader Insight Agency",
+            description="The Loader Insight Agency is a platform that tracks activities of malware loaders through C2 traffic emulation",
+        )
+
+    def generate_relationship(self, source_id, target_id, relation_type="based-on"):
+        return stix2.Relationship(
+            id=StixCoreRelationship.generate_id(relation_type, source_id, target_id),
+            relationship_type=relation_type,
+            source_ref=source_id,
+            target_ref=target_id,
+            created_by_ref=self.identity["standard_id"],
+            # object_marking_refs=[self.tlp],
+        )
+
+    def get_feed_data(self):
+
+        headers = {"Authorization": self.api_key}
+        response = requests.get(self.api_url, headers=headers)
+
+        if response.status_code != 200:
+            self.helper.connector_logger.error(
+                f"Failed to fetch data, status code: {response.status_code}"
+            )
+            return None
+
+        try:
+            return response.json()
+        except ValueError as e:
+            self.helper.connector_logger.error(f"Error parsing JSON response: {e}")
+            return None
+
+    def process_data(self, data):
+
+        stix_objects = []
+        relationships = []
+
+        current_time = datetime.now(dt.UTC)
+
+        lia_external_reference = stix2.ExternalReference(
+            source_name="Loader Insight Agency",
+            url="https://loaderinsight.agency/",
+            description="Loader Insight Agency URL",
+        )
+
+        for item in data["results"]:
+            sha256 = item.get("sha256")
+            filetype = item.get("filetype")
+            first_seen = item.get("first_seen")
+            source_family = item.get("source_family").capitalize()
+            source_botnet = item.get("source_botnet")
+            size = item.get("size")
+            source_url = item.get("source_url")
```

Comment:
Sometimes `source_url` is empty, which lead to some errors during ingestion:
![image](https://github.com/user-attachments/assets/02b3f1a9-925f-4ed9-82e3-771fb28e7da0)

and indicators referencing these URLs are incomplete too:
![image](https://github.com/user-attachments/assets/29ed8c0d-7480-4d78-9e93-da41e0be9682)

At least, in case of missing `source_url` the corresponding stix objects could be skipped (there may be better options though, up to you).

Reply:
As mentioned in a comment further down, this was resolved in the backend. I forgot to resolve this comment :)